build-logic/convention/build.gradle.kts

```diff
@@ -28,6 +28,7 @@ java {
 dependencies {
     compileOnly(libs.android.gradlePlugin)
     compileOnly(libs.kotlin.gradlePlugin)
+    compileOnly(files(libs.javaClass.superclass.protectionDomain.codeSource.location))
```

Comment:
Is this the main line that makes this possible?

Could you provide a link to documentation for how this works? At first glance, it looks a little magical and brittle.

Reply:
This is the workaround for https://github.com/gradle/gradle/issues/15383#issuecomment-779893192.

Reply:
this whole setup is probably too hacky and brittle to show in a sample, as Alex said

I would gladly accept this when it's fully supported by Gradle, it would be great to have proper catalog support in the build-logic